### PR TITLE
samples: USB mass storage sample.yaml cleanup

### PR DESCRIPTION
**Pull request by ajaykish**
Add console harness support for USB mass storage test.
Also, using overlay config for DISK_ACCESS configuration
and accessing the different config file with different
method.

Signed-off-by: Ajay Kishore <ajay.kishore@intel.com>

### CODE REVIEW

**Review by finikorg — CHANGES_REQUESTED** [review]
I wonder could we better use overlay config for `DISK_ACCESS` configuration, we could have 2 tests with different access method, check `samples/net/echo_server/sample.yaml`:

```
  test_usbnet:
    harness: net
    extra_args: OVERLAY_CONFIG="overlay-netusb.conf"
    platform_whitelist: quark_se_c1000_devboard 96b_carbon
    tags: net usb
```

File: samples/subsys/usb/mass/prj.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 CONFIG_STDOUT_CONSOLE=y
 CONFIG_SYS_LOG=y
-CONFIG_SYS_LOG_DEFAULT_LEVEL=1
+CONFIG_SYS_LOG_DEFAULT_LEVEL=3
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/mass/prj.conf]: Please leave SYS_LOG_LEVEL as it was.

**Comment by jfischer-phytec-iot** [comment]
>I wonder could we better use overlay config for DISK_ACCESS configuration, we could have 2 tests with different access method

+1

File: samples/subsys/usb/mass/prj.conf (hunk 1 above)
Comment (galak) [reply samples/subsys/usb/mass/prj.conf]: @ajaykish what does 'internal test automation' mean?  Why doesn't your internal test automation set this to the level it wants?

Comment (nashif) [reply samples/subsys/usb/mass/prj.conf]: > @jfischer-phytec-iot This is required for internal test automation.

This is not a good enough reason, this is a sample, not a test and making changes here does affect everyone using it. Depending on a certain log level in a sample might not provide a solid  test environment.

File: samples/subsys/usb/mass/overlay-flash-disk.conf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#config to disk access over SPI flash
```
[the remark this answers is not in the document]
Comment (ajaykish) [reply samples/subsys/usb/mass/overlay-flash-disk.conf]: Done.

**Comment by ajaykish** [comment]
Thanks @finikorg , as per your suggestion added  the extra_configs in sample.yaml .

**Review by finikorg — APPROVED** [review]

**Review by jfischer-no — COMMENTED** [review]
LGTM, but maybe we should wait for #10168

**Review by nashif — APPROVED** [review]

**Pull request merged**